### PR TITLE
agent-ui: fix segment link

### LINKED ISSUE #272: agent-ui: bad link to segment
### Expected behavior
Display the segment

### Actual behavior
Crash UI and send bad linkhash to agent

### Steps to reproduce the behavior

apr-06-2018 17-32-02

### PR DESCRIPTION
**Pull request by alexppxela**
Resolves #272

---
This change is [Reviewable](https://reviewable.io/reviews/stratumn/js-indigocore/273)

### CODE REVIEW

File: packages/agent-ui/src/containers/topBar.js [hunk 1]
```diff
@@ -23,16 +23,18 @@ const renderTopBarLinks = path => {
   }
 
   const parts = path.split('/').filter(p => p);
+  const partNames = new Map();
   // shorten the segment linkHash in the topbar breadcrumbs display
   if (parts.length === 4 && parts[2] === 'segments') {
-    parts[3] = shortHash(parts[3]);
+    partNames.set(parts[3], shortHash(parts[3]));
```
Comment (simonvadee) [inline packages/agent-ui/src/containers/topBar.js]: why do you need a `Map()` for that ? cannot you do
```javascript
let partNames
 if (parts.length === 4 && parts[2] === 'segments') {
    partNames = shortHash(parts)
}
...
const name = partNames || p;
```

**Comment by t-bast** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/stratumn/js-indigocore/273#-:-L9eX9D87X6-mDRZ7w3f:bnfp4nl)*

**Comment by such** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/stratumn/js-indigocore/273#-:-L9eYf0QEQWtbIGg28ZA:bnfp4nl)*

**Comment by simonvadee** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/stratumn/js-indigocore/273#-:-L9eYnBZ6iE5A32Ik3Sn:bnfp4nl)*

**Pull request merged**